### DISCUSSION TITLE
Number Field supporting thousand separators?

### DISCUSSION THREAD

**Discussion by nicolaidanekov** [microsoft/fluentui-blazor#329] [Q&A] [answered]
Hey there!

Is there any way to configure the Number Field input to support showing thousand separators?
I need a way of automatically adding thousand separators while typing decimals into the input field. 
The bound value would not include the separators, so they would be purely visual.
The intended behavior would be much like in this fiddle:
https://codepen.io/sdarmaputra/pen/OoXmov

Is there already a way of doing this using the Number Field input? :)

**Comment by vnbaaij** [comment, marked as answer]
Hi,

Unfortunately, there is no way of doing this with the Number field. This is not on the components per se but has to do with the underlying HTML `<input type="number">`. Browser support is flaky for that one. For example, you can't specify which character to use as a decimal separator. See also [this link](https://www.ctrl.blog/entry/html5-input-number-localization.html#:~:text=The%20value%20of%20an%20%3Cinput%20type%3D%22number%22%3E%20must%20always,user%20doesn%E2%80%99t%20need%20to%20meet%20this%20technical%20requirement) for more information.
A good resource on [why you should avoid using a number field](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/)

**Reply by nicolaidanekov** [reply]
I thought as much.. :p 
Thanks for the swift reply. ;)
Guessing I'll need to look into using a regular text field in that case. 👍
